website/static/js/registrationUtils.js

```diff
@@ -726,34 +726,55 @@ Draft.prototype.registerWithoutReview = function() {
 Draft.prototype.register = function(url, data) {
     var self = this;
 
-    $osf.block();
-    var request = $osf.postJSON(url, data);
-    request
-        .done(function(response) {
-            if (response.status === 'initiated') {
-                window.location.assign(response.urls.registrations);
+    var nodeIds = data.nodesToRegister.map(function (node) {
+        return node.id;
+    });
+
+    var payload = {
+        'data': {
+            'type': 'registrations',
+            'attributes': {
+                'draft_registration': window.contextVars.draft.pk,
+                'registration_choice': data.registrationChoice,
+                'children': nodeIds
             }
-        })
-        .fail(function() {
-            bootbox.alert({
-                title: 'Registration failed',
-                message: language.registerFail,
-                callback: function() {
-                    $osf.unblock();
-                    if (self.urls.registrations) {
-                        window.location.assign(self.urls.registrations);
-                    }
-                },
-                buttons: {
-                    ok: {
-                        label: 'Back to project',
-                    }
+        }
+    };
+
+    if(data.embargoEndDate){
+        payload.data.attributes.lift_embargo = data.embargoEndDate.format('YYYY-MM-DDThh:mm:ss');
+    }
+
+    $osf.block();
+    var request = $.ajax({
+        type: 'POST',
+        url: url,
+        data: JSON.stringify(payload),
+        contentType: 'application/json',
+        dataType: 'json',
+        xhrconfig: $osf.setXHRAuthorization,
+        xhrFields: {withCredentials: true},
+    }).done(function(response) {
+        window.location.assign(response.data.links.html);
+    }).fail(function(response) {
+        bootbox.alert({
+            title: 'Registration failed',
+            message: language.registerFail,
+            callback: function() {
+                $osf.unblock();
+                if (self.urls.registrations) {
+                    window.location.assign(self.urls.registrations);
                 }
-            });
-        })
-        .always(function() {
-            $osf.unblock();
+            },
+            buttons: {
+                ok: {
+                    label: 'Back to project',
+                }
+            }
         });
+        $osf.unblock();
```

Comment:
Should this be outside of the `fail` function? Seems like this only happens at the end of `fail` where it used to happen in the call to `always`